driver-ui/src/main/java/eu/tsystems/mms/tic/testframework/utils/UITestUtils.java

```diff
@@ -308,32 +308,34 @@ public static List<Screenshot> takeScreenshots() {
     }
 
     /**
-     * Make screenshots from all open browser windows/selenium, webdriver instances of the current thread
-     * and of exclusive webdriver instances of the current method context (if exists).
+     * Make screenshots from all open Webdriver instances the current method context.
      *
      * @param publishToReport True for publish directly into report.
      * @return ScreenshotPaths.
      */
     public static List<Screenshot> takeScreenshots(final boolean publishToReport) {
-        Optional<MethodContext> optionalMethodContext = executionContextController.getCurrentMethodContext();
-
-        // Find exclusive sessions linked to current method context
-        Stream<String> exclusiveSessionKeys = Stream.empty();
-        if (optionalMethodContext.isPresent()) {
-            exclusiveSessionKeys = optionalMethodContext.get().readSessionContexts()
-                    .map(SessionContext::getSessionKey)
-                    .filter(sessionKey -> sessionKey.startsWith(SessionContext.EXCLUSIVE_PREFIX));
+        Optional<MethodContext> methodContext = executionContextController.getCurrentMethodContext();
+
+        if (methodContext.isEmpty()) {
+            LOGGER.warn("Please use this method only in test or setup methods. Otherwise no screenshots are created.");
```

Comment:
In case that the WebDriver was created via Bad Practice (e.g. as a static field), this message would be misleading, would it not? We might call this method "the right way" and still land here, if we made a - from the outside seemingly - unrelated mistake.

Reply:
In case of a bad creation of a WebDriver but correct usage of UiTestUtils the method context is not empty, but there is no session.